superset/utils/core.py

```diff
@@ -184,6 +184,15 @@ class DatasourceType(str, Enum):
     VIEW = "view"
 
 
+class HeaderDataType(TypedDict):
+    notification_format: str
+    owners: List[int]
+    notification_type: str
+    notification_source: Optional[str]
+    chart_id: Optional[int]
+    dashboard_id: Optional[int]
```

Comment:
all of these are optional, right?

Reply:
Type and format shouldn't be since they are required in order to make a report. I can make all of them optional however.